scripts/preprocess-docs.js

```diff
@@ -0,0 +1,144 @@
+
+const docsFilepath = path.join(__dirname, '..', 'docs', 'index.md');
+
+/**
+ * Inserts/updates a table of contents into the documentation
+ * @returns Promise<void>
+ */
+function updateTOC() {
+  return new Promise((resolve, reject) => {
+    console.error(`Updating TOC in ${docsFilepath}...`);
+
+    miss.pipe(
+      fs.createReadStream(docsFilepath, 'utf8'),
+      miss.concat(input =>
+        fs.writeFileSync(
+          docsFilepath,
+          toc
+            .insert(String(input), {
+              bullets: '-',
+              maxdepth: 2
+            })
+            .replace(/\n\n$/, '\n')
+        )
+      ),
+      err => {
+        if (err) {
+          return reject(err);
+        }
+        console.error(`Updated TOC in ${docsFilepath}.`);
+        resolve();
+      }
+    );
+  });
+}
+
+/**
+ * Inserts/updates command-line usage info into the documentation
+ * @returns Promise<void>
+ */
+function updateUsage() {
+  return new Promise((resolve, reject) => {
+    console.error(`Updating command-line usage in ${docsFilepath}...`);
+
+    const FENCE_HEAD = '```plain';
+    const FENCE_FOOT = '```';
+    const TOP_COMMENT = '<!-- usage -->';
+    const BOTTOM_COMMENT = '<!-- usagestop -->';
+    exec(
+      `${process.execPath} ${path.join(
+        __dirname,
+        '..',
+        'bin',
+        'mocha'
+      )} '--help'`,
+      (err, stdout) => {
+        if (err) {
+          return reject(err);
+        }
+        stdout = stdout.trim();
+
+        miss.pipe(
+          fs.createReadStream(docsFilepath, 'utf8'),
+          miss.concat(input => {
+            let trailingNewlines = '';
+            var m = /\n+$/.exec(input);
+            if (m) {
+              trailingNewlines = m[0];
+            }
+            const sections = input
+              .split(/(?:<!--\s+usage(?:\s*stop)?\s+-->)/g)
+              .map(v => v.trim());
+            if (sections.length > 3) {
+              throw new Error('invalid; multiple "usage" sections found');
+            }
+            if (sections.length === 3) {
+              sections.splice(
+                1,
+                1,
+                stripIndents`
+                ${TOP_COMMENT}
+
+                ${FENCE_HEAD}
+                ${stdout}
+                ${FENCE_FOOT}`
+              );
+              sections.splice(2, 0, BOTTOM_COMMENT);
+            } else if (sections.length === 2) {
+              sections.splice(
+                1,
+                0,
+                stripIndents`
+                ${TOP_COMMENT}
+
+                ${FENCE_HEAD}
+                ${stdout}
+                ${FENCE_FOOT}
+                ${BOTTOM_COMMENT}`
+              );
+            } else {
+              throw new Error(
+                `could not find placeholder to update usage in ${docsFilepath}`
+              );
+            }
+
+            fs.writeFileSync(
+              docsFilepath,
+              sections.join('\n\n') + trailingNewlines
+            );
```

Comment:
Would have expected the "Updated usage" message here...

Reply:
the last function to `mississippi.pipe()` is a "finish" handler, which is not necessarily an error condition.